assets/src/checkout/pages/BamboraPayPage.tsx

```diff
@@ -102,20 +161,58 @@ const BamboraPayPage: React.FC<Props> = ({ path, gatewayAccount, payment }) => {
     expiry: generateId("card-expiry-")
   });
 
+  const [fieldComplete, setFieldComplete] = React.useState<null | {
+    field: Field;
+  }>(null);
+  const [fieldError, setFieldError] = React.useState<null | {
+    field: Field;
+  }>(null);
   const [cardErrors, setCardErrors] = React.useState<Record<Field, string>>({
     "card-number": "",
     cvv: "",
     expiry: ""
   });
+  const [cardBrand, setCardBrand] = React.useState<null | Brand>(null);
+  const cardTypesQuery = useCardTypesQuery({ errorPolicy: "all" });
 
   const onCheckoutLoad = React.useCallback(
     (checkout: CustomCheckout) => {
       mountFields(checkout, theme, cardFieldIds);
-      bindEventListeners(checkout, setCardErrors);
+      bindEventListeners(
+        checkout,
+        setFieldComplete,
+        setFieldError,
+        setCardErrors,
+        setCardBrand
+      );
     },
     [theme, cardFieldIds]
   );
 
+  React.useEffect(() => {
+    if (fieldError) {
+      return;
+    }
+    if (!cardBrand || isCardBrandSupported(gatewayAccount, cardBrand)) {
+      setCardErrors(errors => ({ ...errors, "card-number": "" }));
+      return;
+    }
+    if (cardTypesQuery.loading || !cardTypesQuery.data) {
+      return;
+    }
+    const bamboraBrand = normalizeBamboraBrand(cardBrand);
+    const cardType = cardTypesQuery.data.cardTypes.find(
+      cardType => cardType.brand === bamboraBrand
+    );
+
+    setCardErrors(errors => ({
+      ...errors,
+      "card-number": !cardType
+        ? "This card is not supported"
```

Comment:
@acodegirl missing `.` to be consistent

Reply:
I reverted import /auth & . changes by mistake. Just pushed the changes.